queries/tags.scm

```diff
@@ -0,0 +1,54 @@
+; Modules
+(module) @name @reference.module
+(import alias: (identifier) @name) @reference.module
+(remote_type_identifier
+  module: (identifier) @name) @reference.module
+((field_access
+  record: (identifier) @name)
+ (#is-not? local)) @reference.module
+
+; Functions
+(function
+  name: (identifier) @name) @definition.function
+(public_function
+  name: (identifier) @name) @definition.function
+(external_function
+  name: (identifier) @name) @definition.function
+(public_external_function
+  name: (identifier) @name) @definition.function
+(unqualified_import (identifier) @name) @reference.function
+((function_call
+   function: (identifier) @name) @reference.function
+ (#is-not? local))
+((field_access
+  record: (identifier) @ignore
+  field: (label) @name)
+ (#is-not? local)) @reference.function
+((binary_expression
+   operator: "|>"
+   right: (identifier) @name)
+ (#is-not? local)) @reference.function
+
+; Types
+(public_type_definition
+  (type_name
+    name: (type_identifier) @name)) @definition.type
+(type_definition
+  (type_name
+    name: (type_identifier) @name)) @definition.type
+(public_type_definition
+  (type_constructors
+    (type_constructor
+      name: (type_identifier) @name))) @definition.type
+(type_definition
+  (type_constructors
+    (type_constructor
+      name: (type_identifier) @name))) @definition.type
```

Comment:
Hmm yeah I wasn't sure about this. I think this ends up working out well because you can jump between usages of data constructors and their definitions. The suffix in `definition.<kind>` or `reference.<kind>` doesn't have to be `module|function|type`, we could distinguish these as `data_constructor`. OTOH it works out nicely to just label all remaining `type_identifiers` as `reference.type`. I'm not sure what the best move is :upside_down_face:

Reply:
I think for now we'll keep it as-is :+1: